### DISCUSSION TITLE
How do I validate a single type for multiple constraints?

### DISCUSSION THREAD

**Discussion by i-am-the-slime** [Iltotore/iron#247] [Q&A] [answered]
I have this currently:

```scala
(oib.refineNec[Numeric], oib.refineNec[FixedLength[11]]).parMapN(
        (_, _)  => oib.asInstanceOf[PlatformOib]
      )
```

Surely, there's a better way!

**Comment by i-am-the-slime** [comment]
I'm also surprised I can't do `refineUnsafe` which I thought would be `asInstanceOf` really.

**Comment by Iltotore** [comment, marked as answer]
Hello.

Can you provide the definition of your `PlatformOib`?

Anyway I think you are looking for `refineFurther` variants. There is an example in the docs IIRC.

When you just want to cast a raw value to its constrained form without actually checking the constraint, you should use `assume`.

**Reply by i-am-the-slime** [reply]
Thanks! I'll try with `refineFurther`

**Reply by Iltotore** [reply]
👍

If you solved your issue, feel free to mark the message as answer so this thread will be marked as solved.

**Reply by i-am-the-slime** [reply]
I'm fighting with something else already, so I'll close this.
